src/AutoMapper/Execution/PropertyAccessor.cs

```diff
@@ -1,24 +1,28 @@
+using System.Linq.Expressions;
+
 namespace AutoMapper.Execution
 {
     using System;
     using System.Reflection;
 
-    public class PropertyAccessor : PropertyGetter, IMemberAccessor
+    public class PropertyAccessor<TSource, TValue> : PropertyGetter<TSource, TValue>, IMemberAccessor
     {
-        private readonly Lazy<LateBoundPropertySet> _lateBoundPropertySet;
+        private readonly Lazy<Expression<LateBoundPropertySet<TSource, TValue>>> _lateBoundPropertySetExpression;
+        private readonly Lazy<LateBoundPropertySet<TSource, TValue>> _lateBoundPropertySet;
 
         public PropertyAccessor(PropertyInfo propertyInfo)
             : base(propertyInfo)
         {
             var HasSetter = propertyInfo.GetSetMethod(true) != null;
-            _lateBoundPropertySet = HasSetter
-                ? new Lazy<LateBoundPropertySet>(() => DelegateFactory.CreateSet(propertyInfo))
-                : new Lazy<LateBoundPropertySet>(() => (_, __) => { });
+            _lateBoundPropertySetExpression = HasSetter
+                ? new Lazy<Expression<LateBoundPropertySet<TSource, TValue>>>(() => DelegateFactory.CreateSet<TSource, TValue>(propertyInfo))
```

Comment:
These types with multiple angle brackets could be made more readable by wrapping them inside a struct at zero cost.